src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.KeyTrans.cs

```diff
@@ -142,20 +140,42 @@ internal byte[] DecryptCek(X509Certificate2 cert, RSA privateKey, out Exception
                     recipient.RecipientIdentifierType.ToString());
             }
 
-            RSAEncryptionPadding padding;
+            RSAEncryptionPadding padding = recipient.RSAEncryptionPadding ?? recipient.Certificate.GetKeyAlgorithm() switch {
```

Comment:
Please don't use the new switch syntax.  Crypto code should definitely favor legibility over coolness.

Reply:
(and, yes, I think that

```C#
RSAEncryptionPadding padding = recipient.RSAEncryptionPadding;

if (padding == null)
{
    if (recipient.Certificate.GetKeyAlgorithm() == Oids.RsaOaep)
    {
        padding = RSAEncryptionPadding.OaepSha1;
    }
    else
    {
        padding = RSAEncryptionPadding.Pkcs1;
    }
}
```

is **much** more legible.)

Reply:
To each their own :-). Will change.